nixos/modules/services/misc/geoipupdate.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.geoipupdate;
+  configFile = pkgs.writeText "geoipupdate.conf" ''
+    AccountID ${cfg.accountID}
+    LicenseKey ${cfg.licenseKey}
+    EditionIDs ${concatStringsSep " " cfg.editionIDs}
+    ${concatStringsSep "\n" (mapAttrsToList (k: v: "${k} ${v}") cfg.extraConfig)}
+  '';
+in {
+  options.services.geoipupdate = with types; {
+    enable = mkEnableOption "the automated GeoIP updater";
+
+    accountID = mkOption {
+      description = "Account ID of your Maxmind account (or 0 for free GeoLite DBs)";
+      type = str;
+      default = "0";
+    };
+
+    licenseKey = mkOption {
+      description = "License key of your Maxmind account (or 000000000000 for free GeoLite DBs)";
+      type = str;
+      default = "000000000000";
+    };
+
+    editionIDs = mkOption {
+      description = "Edition IDs to update";
+      type = listOf str;
+      default = [ "GeoLite2-Country" "GeoLite2-City" ];
+    };
+
+    extraConfig = mkOption {
+      description = "Extra configuration to append to the configuration file";
+      type = attrsOf str;
+      default = {};
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.geoipupdate = {
+      description = "Maxmind GeoIP updater";
+      after = [ "networking.target" ];
+      wantedBy = [ "multi-user.target" ];
+      startAt = "weekly";
```

Comment:
`RandomizedDelaySec` for the timer might be useful to avoid DDoSing the Maxmind servers. The original service uses 1 hour.